### ISSUE TITLE
Feature Request: option to blur all GTK4 windows

### ISSUE THREAD

**Issue by joaocandre** [aunetx/blur-my-shell#604]
If possible, instead of handpicking the windows to blur, allow grouping all GTK4 windows/apps in a single setting.

Also, an option to disable blur on maximized windows (which is already the behavior with GTK4 apps, but not, for instance, with Firefox and Thunderbird)

**Comment by FelixFourcolor** [comment]
> Also, an option to disable blur on maximized windows
Fyi, blur is already disabled for fullscreen windows, which is very useful for me. Idk how useful it is to have it for maximized windows as well.

> which is already the behavior with GTK4 apps
Wait what? I've never seen such thing. How do you enable it?

**Comment by joaocandre** [comment]
> Fyi, blur is already disabled for fullscreen windows. I don't think it's necessary to have it for maximized windows as well.

Not in Mozilla apps, they get blurred when maximized when app blurring is enabled. At least in my setup.

> Wait what? I've never seen such thing. How do you enable it?

From what I can see, all Gnome apps get un-blurred when maximized, I assume that's the default behavior.

**Comment by FelixFourcolor** [comment]
Oh I see what's going on. You're using X11 right? On Wayland all fullscreen apps are unblurred (not maximized, but fullscreen, like when you fullscreen a youtube video). It's really useful, try it out!

(I don't know why this is only available in wayland, and it's not documented either, I should probably open an issue about it.)

I don't know any app that gets unblurred when maximized though, on either X11 or Wayland. That sounds like a bug to me. But anyways, just use Wayland, then you can go fullscreen to unblur apps and you don't need to worry about this anymore.

**Comment by joaocandre** [comment]
No, I'm pretty sure I'm using Wayland. Maybe what I am describing is specific to Firefox and Thunderbird, but I don't think it has to do with full screen, but rather maximized windows. It occurs whether I hand-pick the apps to blur or use the 'blur all apps' option:

[Screencast from 2024-06-07 23-09-08.webm](https://github.com/aunetx/blur-my-shell/assets/1812344/9f340f44-4ac3-43a3-b598-31db78d2f2ed)

> all fullscreen apps are unblurred (not maximized, but fullscreen

That may be, but in my (admittedly anecdotal) use case, maximized windows usually require full focus and blur is often not desired. For a while I saw it happen with nautilus (maybe between updates) but indeed it does not happen anymore (although that was the case when I opened this issue).

**Comment by FelixFourcolor** [comment]
> On Wayland all fullscreen apps are unblurred (not maximized, but fullscreen, like when you fullscreen a youtube video). It's really useful, try it out!
>
> (I don't know why this is only available in wayland, and it's not documented either, I should probably open an issue about it.)

I just opened the issue #624. Can you replicate it?

**Comment by aunetx** [comment]
Hello, that could be quite neat to some extent, but I don't think I can detect GTK4 (or anything technical about the app whatsoever, except if they use wayland or xorg/xwayland) from gnome-shell. So it is a technical no for the moment...
